test/clj/rems/api/test_audit_log.clj

```diff
@@ -0,0 +1,110 @@
+          (testing "> GET"
+            (testing "> unauthorized"
+              (is (response-is-forbidden? (-> (request :get "/api/users/active")
+                                              (authenticate "42" "alice")
+                                              handler))))
+            (testing "> authorized"
+              (is (= 200 (:status (-> (request :get "/api/users/active")
+                                      (authenticate "42" "owner")
+                                      handler))))))
+          (reset! time-a (time/now))
+          (testing "> POST"
+            (testing "> status 200, different api key"
+              (is (false? (:success (-> (request :post "/api/applications/submit")
+                                        (authenticate "43" "alice")
+                                        (json-body {:application-id 99999999999})
+                                        handler
+                                        read-ok-body)))))
+            (testing "> status 400"
+              (is (response-is-bad-request? (-> (request :post "/api/applications/submit")
+                                                (authenticate "42" "alice")
+                                                (json-body {:boing "3"})
+                                                handler))))
+            (testing "> status 500"
+              (with-redefs [rems.api.services.command/command! (fn [_] (throw (Error. "BOOM")))]
+                (is (response-is-server-error? (-> (request :post "/api/applications/submit")
+                                                   (authenticate "42" "alice")
+                                                   (json-body {:application-id 3})
+                                                   handler)))))))
+        (testing "> session"
+          (let [cookie (login-with-cookies "malice")
+                csrf (get-csrf-token cookie)]
+            (testing "> GET"
+              (is (= 200 (:status (-> (request :get "/api/catalogue")
+                                      (header "Cookie" cookie)
+                                      (header "x-csrf-token" csrf)
+                                      handler)))))
+            (testing "> failed PUT"
+              (is (response-is-forbidden? (-> (request :put "/api/catalogue-items/archived")
+                                              (header "Cookie" cookie)
+                                              (header "x-csrf-token" csrf)
+                                              (json-body {:id 9999999 :archived true})
+                                              handler))))))))
+    (testing "user can't get log"
+      (is (response-is-forbidden? (-> (request :get "/api/audit-log")
+                                      (authenticate "42" "alice")
+                                      handler))))
+    (testing "reporter can get log"
+      (is (= [{:userid nil :apikey nil :method "get" :path "/api/unknown" :status "404"}
```

Comment:
One of the acceptance test cases is that can the reporter see who has viewed a specific application? I don't think this is covered here yet. Moreover, isn't it a requirement that someone can figure out who has seen someone's data? How would one figure out which applications a person is in, and then filter from this log all the people who have seen such an application? That could be its own API I guess.

Reply:
Yeah some further work on querying might be needed. Let's discuss and make a new issue? I think this feature makes sense to merge already in this state.

Reply:
Ok as long as the concern is in some task on the board.

Reply:
I created #2147, and added filtering by application to this PR